### PR TITLE
Code Review: Update LibraryType enum (#684)

### PR DESCRIPTION
**Pull request by nthState**
Resolves: #684

### CODE REVIEW

**Comment by bohemian-coding-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>This pull request may need a CHANGELOG entry.</td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  `Source/dom/models/Library.js` changed, but not:
  - 📚 its docs

That's OK as long as you're refactoring.
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against b3af86b3a664698cd37dd2af70b6d9cadf53e7ad
</p>

File: Source/dom/models/Library.js [hunk 1]
```diff
@@ -16,14 +16,18 @@ const AddStatus = {
...
+  2: 'RemoteUser', // Previously `Remote`, is this a library that can be updated using an appcast.
+  3: 'RemoteTeam', // Is this a library hosted on Sketch for Teams
+  4: 'RemoteThirdParty', // Remote 3rd Party
 }
 
 const LibraryType = {
   Internal: 'Internal',
-  User: 'User',
-  Remote: 'Remote',
+  LocalUser: 'LocalUser', // Previously `User`
```
Comment (mathieudutour) [inline Source/dom/models/Library.js]: That’s a breaking change. 
Consider keeping the two previous enum cases while remapping them to their new value (and perhaps show a deprecation message)

File: Source/dom/models/Library.js [hunk 2]
```diff
@@ -16,14 +16,20 @@ const AddStatus = {
...
+  2: 'RemoteUser',
+  3: 'RemoteTeam',
+  4: 'RemoteThirdParty',
 }
 
 const LibraryType = {
   Internal: 'Internal',
-  User: 'User',
-  Remote: 'Remote',
+  User: 'LocalUser', // Mapping `User` to new value `LocalUser`
```
Comment (christianklotz) [inline Source/dom/models/Library.js]: Let's make this and `Remote` super obvious that they shouldn't be used going forward. Something like this perhaps? 🤔 

```suggestion
  // @deprecated Please use `LocalUser` instead
  User: 'LocalUser', // This value has been remapped for compatibility and will be removed in the future.
```

**Comment by jedrichards** [comment]
Closes in favour of https://github.com/sketch-hq/SketchAPI/pull/703

**Pull request closed without merging**